helix-term/src/ui/editor.rs

```diff
@@ -669,12 +669,14 @@ impl EditorView {
 
         let mut lines = Vec::new();
         for diagnostic in diagnostics {
-            let style = Style::reset().patch(match diagnostic.severity {
-                Some(Severity::Error) => error,
-                Some(Severity::Warning) | None => warning,
-                Some(Severity::Info) => info,
-                Some(Severity::Hint) => hint,
-            });
+            let style = Style::default().remove_modifier(Modifier::all()).patch(
```

Comment:
Would it be better if we patch it twice after reset? First with normal text background, then with diagnostic style.

Reply:
I'm not sure I understand. This is intended to fix the reported problem by removing the modifiers for the diagnostics (prior to applying the diagnostic style, which can include modifiers) while leaving the other style elements untouched.

What would patching twice buy us here?

Reply:
Yeah we could do

```rust
Style::reset().patch(background_style).patch(
  match {
    // ... that existing match expression ...
  },
)
```

I'm pretty sure that would be correct - I don't think there are any odd scenarios where we would want inherit a background that isn't the regular background, right 🤔?

Reply:
Ah, I understand. OK. One sec.

Reply:
`reset.patch(background)` should be identical to just using `background` though.

Reply:
Looking at `Style::reset`, it looks like it'll start out with a `sub_modifier` of `Modifier::all()` which I think is what we need to clear the italics. We could probably do:

```rust
background_style.remove_modifier(Modifier::all())
```

to be more clear that we only want to drop the modifiers from the background

Reply:
Did you want me to follow up with this change, or are we all good at this point?

Reply:
This looks like it's working well for me now so I say no need for any more changes 👍

Reply:
Copy. Thanks, and sorry for the original oversight.